### PR TITLE
Improve the check for Vulkan to check for surface extensions first.

### PR DESCRIPTION
**Pull request by hrydgard**
Might fix the crashing issues we've seen on AMD?

### CODE REVIEW

File: Common/Vulkan/VulkanLoader.cpp [hunk 1]
```diff
@@ -260,18 +261,52 @@ bool VulkanMayBeAvailable() {
...
+	const char * const surfaceExtension = VK_KHR_WIN32_SURFACE_EXTENSION_NAME;
 #elif defined(__ANDROID__)
-	instanceExtensions[0] = VK_KHR_ANDROID_SURFACE_EXTENSION_NAME;
-	ci.enabledExtensionCount = 1;
+	const char *surfaceExtension = VK_KHR_ANDROID_SURFACE_EXTENSION_NAME;
+#else
+	const char *surfaceExtension = 0;
 #endif
 
+	if (surfaceExtension) {
```
Comment (unknownbrackets) [inline Common/Vulkan/VulkanLoader.cpp]: if !, it's Linux so we're not checking here?  I guess that's hopefully okay.  I wonder how the Linux AMD drivers are...

-[Unknown]

File: Common/Vulkan/VulkanLoader.cpp [hunk 2]
```diff
@@ -260,18 +261,51 @@ bool VulkanMayBeAvailable() {
...
 #ifdef _WIN32
-	instanceExtensions[0] = VK_KHR_WIN32_SURFACE_EXTENSION_NAME;
-	ci.enabledExtensionCount = 1;
+	const char * const surfaceExtension = VK_KHR_WIN32_SURFACE_EXTENSION_NAME;
 #elif defined(__ANDROID__)
-	instanceExtensions[0] = VK_KHR_ANDROID_SURFACE_EXTENSION_NAME;
-	ci.enabledExtensionCount = 1;
+	const char *surfaceExtension = VK_KHR_ANDROID_SURFACE_EXTENSION_NAME;
+#else
+	const char *surfaceExtension = 0;
```
Comment (unknownbrackets) [inline Common/Vulkan/VulkanLoader.cpp]: nit: nullptr?

-[Unknown]

**Review by unknownbrackets — COMMENTED** [review]
Hopefully this stops the crashing indeed.

-[Unknown]

File: Common/Vulkan/VulkanLoader.cpp (hunk 1 above)
Comment (hrydgard) [reply Common/Vulkan/VulkanLoader.cpp]: I just didn't want to change the flow on other platforms. I'll fix it up for Linux later.

File: Common/Vulkan/VulkanLoader.cpp (hunk 2 above)
Comment (hrydgard) [reply Common/Vulkan/VulkanLoader.cpp]: Yeah, oops, old instinct took over somehow.

**Pull request merged**